script.js

```diff
@@ -0,0 +1,209 @@
+    return date.toTimeString().split('').splice(0, 5).join('');
+}
+
+function getDate(date) {
+    return date.toLocaleDateString();
+}
+
+function generateId() {
+    return Math.random().toString(36).slice(2);
+}
+
+function getTasksCount() {
+    const current = allTasks.filter(task => !task.isDone).length;
+    const completed = allTasks.length - current;
+    return {current, completed};
+}
+
+function getTaskClassName(priority) {
+    if (priority === PRIORITIES.low) {
+        return TASK_CLASSNAMES.greenFont;
+    } else if (priority === PRIORITIES.medium) {
+        return TASK_CLASSNAMES.yellowFont;
+    } else {
+        return TASK_CLASSNAMES.redFont;
+    }
+}
+addTaskButton.addEventListener('click',()=>{
+    modalLabelText.textContent = TEXT_VARIABLES.addTask;
+    submitButton.textContent = TEXT_VARIABLES.addTask;
+    mediumPriority.checked = true;
+    inputTitle.value = '';
+    inputText.value = '';
+    modalEditMode = false;
+})
+function addTaskLayout(task) {
+    const taskLayout = document.createElement('li');
+    const time = new Date(task.time);
+    taskLayout.className = `list-group-item d-flex w-100 mb-2 ${getTaskClassName(task.priority)}`;
+    taskLayout.innerHTML = `<div class="w-100 mr-2">
+                <div class="d-flex w-100 justify-content-between">
+                    <h5 class="mb-1">${task.title}</h5>
+                    <div>
+                        <small class="mr-2">${task.priority} priority</small>
+                        <small>${getTime(time)} ${getDate(time)}</small>
+                    </div>
+                </div>
+                <p class="mb-1 w-100">${task.text}</p>
+            </div>
+            <div class="dropdown m-2 dropleft">
+                <button class="btn btn-secondary h-100" type="button" id="dropdownMenuItem1"
+                        data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" >
+                    <i class="fas fa-ellipsis-v"></i>
+                </button>
+                <div class="dropdown-menu p-2 flex-column" aria-labelledby="dropdownMenuItem1" id="${task.id}">
+                    ${!task.isDone ? `<button type="button" class="btn btn-success w-100">Complete</button>
+    <button type="button" class="btn btn-info w-100 my-2" data-toggle="modal" data-target="#exampleModal">Edit</button>`
+        : `<button type="button" class="btn btn-success w-100 mb-2">Uncomplete</button>`}
+                    <button type="button" class="btn btn-danger w-100">Delete</button>
+                </div>
+            </div>`;
+    if (sortUp) {
+        task.isDone ? completedTasksBlock.append(taskLayout) : currentTasksBlock.append(taskLayout);
+    } else {
+        task.isDone ? completedTasksBlock.prepend(taskLayout) : currentTasksBlock.prepend(taskLayout);
+    }
+}
+
+function renderApp() {
+    currentTasksBlock.innerHTML = null;
+    completedTasksBlock.innerHTML = null;
+    allTasks.forEach(task => addTaskLayout(task));
```

Comment:
const completedTasksList = []
const tasksList = []
allTasks.forEach((item) => item.completed ? completedTasksList.push(item) : tasksList.push(item))
currentTasksBlock.append/prepend(tasksList)
completedTasksBlock.appnd/prepend(completedTasksList)

Reply:
```suggestion
function renderApp() {
    currentTasksBlock.innerHTML = null;
    completedTasksBlock.innerHTML = null;
    const completedTasksList = allTasks.filter(task => task.isDone).map(task => getTaskLayout(task));
    const currentTasksList = allTasks.filter(task => !task.isDone).map(task => getTaskLayout(task));
    completedTasksBlock.innerHTML = sortUp ? completedTasksList.join('') : completedTasksList.reverse().join('');
    currentTasksBlock.innerHTML = sortUp ? currentTasksList.join('') : currentTasksList.reverse().join('');
    const tasksCount = getTasksCount();
    currentTasksTitle.textContent = `ToDo (${tasksCount.current})`;
    completedTasksTitle.textContent = `Completed (${tasksCount.completed})`;
}
```

Reply:
And function getTaskLayout now looks: 
```suggestion
function getTaskLayout(task) {
    const time = new Date(task.time);
    return `<li class="list-group-item d-flex w-100 mb-2 ${getTaskClassName(task.priority)}">
                <div class="w-100 mr-2">
                    <div class="d-flex w-100 justify-content-between">
                        <h5 class="mb-1">${task.title}</h5>
                        <div>
                            <small class="mr-2">${task.priority} priority</small>
                            <small>${getTime(time)} ${getDate(time)}</small>
                        </div>
                    </div>
                    <p class="mb-1 w-100">${task.text}</p>
                </div>
                <div class="dropdown m-2 dropleft">
                    <button class="btn btn-secondary h-100" type="button" id="dropdownMenuItem1"
                        data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" >
                        <i class="fas fa-ellipsis-v"></i>
                    </button>
                    <div class="dropdown-menu p-2 flex-column" aria-labelledby="dropdownMenuItem1" id="${task.id}">
                        ${!task.isDone ? `<button type="button" class="btn btn-success w-100">Complete</button>
                    <button type="button" class="btn btn-info w-100 my-2" data-toggle="modal" data-target="#exampleModal">Edit</button>`
                    : `<button type="button" class="btn btn-success w-100 mb-2">Uncomplete</button>`}
                    <button type="button" class="btn btn-danger w-100">Delete</button>
                    </div>
                </div>
            </li>`;
}
```